### ISSUE TITLE
Material Canvas/Editor: Tools crash after changing Render Pipeline settings

### ISSUE THREAD

**Issue by LB-KacperMetelski** [o3de/o3de#14596]
**Describe the bug**
Fatal error occurs after changing the Render Pipeline in the Material Canvas' or Material Editor's Viewport.

Multiple asserts are printed in the attached logs with the following one occurring right before the crash:

```
Trace::Assert
 C:/o3de/Gems/Atom/RHI/DX12/Code/Source/RHI/DX12.cpp(267): (9340) 'bool __cdecl AZ::DX12::AssertSuccess(long)'
HRESULT not a success 80070057
```

**Steps to reproduce**
1. Open the Material Canvas or Material Editor.
2. Resize the Viewport to the right.
3. Change the Render Pipeline to any of the three options at the bottom of the list (i.e. Multi View Render Pipeline).

**Expected behavior**
The Render Pipeline is changed.

**Actual behavior**
The Material Canvas/Editor crashes after displaying the fatal error:

"GPU Device was removed. Check the log file for more detail."

**Visual Studio Call stack**
```
>	MaterialCanvas.exe!AZ::Debug::Trace::Crash() Line 290	C++
 	Atom_RPI.Editor.dll!AZ::RPI::RPISystemComponent::OnDeviceRemoved(AZ::RHI::Device * device) Line 194	C++
 	[Inline Frame] Atom_RHI_DX12.Private.dll!AZStd::Internal::INVOKE(void(AZ::RHI::RHISystemNotificiationInterface::*)(AZ::RHI::Device *) &) Line 177	C++
 	[Inline Frame] Atom_RHI_DX12.Private.dll!AZStd::invoke(void(AZ::RHI::RHISystemNotificiationInterface::*)(AZ::RHI::Device *) &) Line 54	C++
 	[Inline Frame] Atom_RHI_DX12.Private.dll!AZ::EBusEventProcessingPolicy::Call(void(AZ::RHI::RHISystemNotificiationInterface::*)(AZ::RHI::Device *) &) Line 437	C++
 	Atom_RHI_DX12.Private.dll!AZ::Internal::EBusContainer<AZ::RHI::RHISystemNotificiationInterface,AZ::RHI::RHISystemNotificiationInterface,0,1>::Dispatcher<AZ::EBus<AZ::RHI::RHISystemNotificiationInterface,AZ::RHI::RHISystemNotificiationInterface>>::Broadcast<void (__cdecl AZ::RHI::RHISystemNotificiationInterface::*)(AZ::RHI::Device *),AZ::RHI::Device *>(void(AZ::RHI::RHISystemNotificiationInterface::*)(AZ::RHI::Device *) && func, AZ::RHI::Device * && <args_0>) Line 1360	C++
 	Atom_RHI_DX12.Private.dll!AZ::RHI::Device::SetDeviceRemoved() Line 124	C++
 	Atom_RHI_DX12.Private.dll!AZ::DX12::Device::OnDeviceRemoved() Line 758	C++
```

**Screenshots/Video**
https://user-images.githubusercontent.com/110163473/219040177-8e2d3561-afda-4a8b-8a88-4113b1ec9568.mp4

**Found in Branch**
Development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
f829a06

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version: 11 22H2
 - CPU: I7-11700f
 - GPU: Geforce RTX3070 MSI
 - Memory: 32GB

**Additional context**
[MaterialCanvas.log](https://github.com/o3de/o3de/files/10743413/MaterialCanvas.log)
[MaterialEditor.log](https://github.com/o3de/o3de/files/10743415/MaterialEditor.log)

**Comment by galibzon** [comment]
@VickyAtAZ

**Comment by moudgils** [comment]
Does this crash happen if you wait for Asset Processor to finish processing all the assets?

**Comment by moudgils** [comment]
https://github.com/o3de/o3de/pull/15439 should fix this

**Comment by LB-MateuszSzweda** [comment]
Issue verified as fixed on Stabilization/2305 (392d44b) for both Material Canvas and Material Editor. Changing Render Pipeline no longer crashes.
